sklearn/svm/_classes.py

```diff
@@ -37,9 +54,15 @@ class LinearSVC(LinearClassifierMixin, SparseCoefMixin, BaseEstimator):
         square of the hinge loss. The combination of ``penalty='l1'``
         and ``loss='hinge'`` is not supported.
 
-    dual : bool, default=True
+    dual : "auto" or bool, default=True
         Select the algorithm to either solve the dual or primal
         optimization problem. Prefer dual=False when n_samples > n_features.
+        `dual="auto"` will choose the value of the parameter automatically,
+        based on the values of `n_samples`, `n_features`, `loss`
+        and `penalty`.
```

Comment:
we should explain exactly how this is done, either here, or in the user guide.

Reply:
I'm not sure how it could be done. We are checking if N>M after that we reuse svm function to check available solvers for chosen loss-penalty-ovr combinations. The problem is in explaining the latter one

Reply:
Added short explanation to docs